src/idom/utils.py

```diff
@@ -49,12 +53,8 @@ def __repr__(self) -> str:
         return f"{type(self).__name__}({current})"
 
 
-_ModelTransform = Callable[[Dict[str, Any]], Any]
-
-
-def html_to_vdom(source: str, *transforms: _ModelTransform) -> Dict[str, Any]:
+def html_to_vdom(html: Union[str, etree._Element], *transforms: _ModelTransform):
```

Comment:
To avoid exposing the underlying implementation (i.e. the `etree._Element` type), maybe there could be the public `html_to_vdom` which accepts just a string and a private `_html_to_vdom` function that does the actual recursive part of the logic.